x-pack/solutions/observability/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -298,6 +302,22 @@ export const createCustomThresholdExecutor = ({
     alertsClient.setAlertLimitReached(hasReachedLimit);
     const recoveredAlerts = alertsClient.getRecoveredAlerts() ?? [];
 
+    let groupingObjectForRecovered: Record<string, object> = {};
+
+    // extracing group by fields from kibana.alert.rule.params,
+    // since all recovered alert documents will have same group by fields,
+    // we are only checking first recovered alert document
+    if (recoveredAlerts.length > 0) {
+      const alertHit = recoveredAlerts[0].hit;
+      const ruleParamsOfRecoveredAlert = alertHit?.[ALERT_RULE_PARAMETERS] as EvaluatedRuleParams;
+      const groupByFields = ruleParamsOfRecoveredAlert?.groupBy;
+
+      groupingObjectForRecovered = getGroupByObject(
```

Comment:
Is this object going to be:

key: alertId
value: groupObject

Nit: Maybe we can add this as a comment. (i.e., // { alertId:  { groupField1: groupValue1 }})

Reply:
As we discussed, another option would be to use the `kibana.alert.group` field in the custom threshold rule. This can simplify the logic for group object creation, but up to you which one to choose.